pkg/proxy/accesslog/record.go

```diff
@@ -244,6 +250,12 @@ type LogRecordKafka struct {
 	Topic KafkaTopic
 }
 
+// LogRecordDNS contains the DNS specific portion of a log record
+type LogRecordDNS struct {
+	// Msg is the DNS message as parsed by the proxy
+	Msg *dns.Msg
```

Comment:
Is this a structure with a stable API? I don't feel overall confident encoding a struct from an external dependency into an API.

Reply:
It definitely isn't guaranteed to be stable. The types have no json annotations, so they will follow the names of the public fields. Looking at the history of `https://github.com/miekg/dns/types.go` until the start of 2015 shows only 1 case where a field in a RR changed type, 1 case where some fields were added and one removed, and a few where whole RR types were added or removed outright.
I'm surprised by how stable it has been but your point is valid. We can:
- Include only A/AAAA responses in the LogRecordDNS data, using a type we define (we already have one that might work). I expect we will want to capture other DNS data, even if it isn't directly used, however.
- We can mirror the whole set of types ourselves. This would also need some code to convert between the two systems.
- We can include the A/AAAA responses in a type we control, in addition to the unmanaged dns.Msg type we have now
- We can pretty print the DNS message into the bind/dig style. That is likely more stable.


I'll include the A/AAAA response data since all the changes above require it, and remove the dns.Msg field since we can always add it later.